internal/recovery/helpers.go

```diff
@@ -26,43 +22,36 @@ func uniquePublications(s []*protocol.Publication) []*protocol.Publication {
 }
 
 // MergePublications ...
-func MergePublications(recoveredPubs []*protocol.Publication, bufferedPubs []*protocol.Publication) ([]*protocol.Publication, bool) {
+func MergePublications(recoveredPubs []*protocol.Publication, bufferedPubs []*protocol.Publication, isLegacyOrder bool) ([]*protocol.Publication, bool) {
 	if len(bufferedPubs) > 0 {
 		recoveredPubs = append(recoveredPubs, bufferedPubs...)
 	}
 	sort.Slice(recoveredPubs, func(i, j int) bool {
-		if recoveredPubs[i].Gen != recoveredPubs[j].Gen {
-			return recoveredPubs[i].Gen > recoveredPubs[j].Gen
+		if isLegacyOrder {
```

Comment:
We can move this `if` out of sort. perf++, readability++

Reply:
fixed (if I understood this point right)